### DISCUSSION TITLE
Prevent transitioning?

### DISCUSSION THREAD

**Discussion by fbloemhof** [basnijholt/adaptive-lighting#912] [Q&A]
In some situations, by example my toilet, I am seeing some unwanted behavior. The last time the lights were on it was mid day so they were on full brightness and high kelvin. When they again get triggered later on in the evening, the light initially turns on at this state and then transitions to the state as AL wants it.

Is there a way to prevent this?

I am using Zigbee2MQTT if this matters.

**Comment by basnijholt** [comment]
Do you have `separate_turn_on_commands` enabled? If so, this is the expected behaviour.

**Comment by fbloemhof** [comment]
Nope, that is disabled. I can repro it with the default config.

27 jan 2024 21:40:45 Bas Nijholt ***@***.***>:

**Comment by basnijholt** [comment]
Then perhaps your light doesn't allow to turn on with arbitrary settings. You can try this out by turning it off and then calling `light.turn_on` with different settings than what it had before it turned off.

**Comment by fbloemhof** [comment]
Hm indeed, same behavior. Then it perhaps is an Z2M issue.

27 jan 2024 22:31:04 Bas Nijholt ***@***.***>:

**Comment by fbloemhof** [comment]
When I bypass the Z2M group the behavior differs. It still seems a little off but it is better then using the Z2M group. Will do some more testing later.

27 jan 2024 22:38:26 Freek  ***@***.***>:

**Comment by fbloemhof** [comment]
With an HA light group the behavior is the same. Only when I call a single light it seems ok (but I still can see some transitioning).

**Comment by fbloemhof** [comment]
Okay considdering my test results in https://github.com/Koenkk/zigbee2mqtt/discussions/21081, it seems a Zigbee thing.

Is it possible to work this around in AL by intercepting the light.turn_off and setting some values somewhere in the middle? From bright to lower bright is more annoying than in reverse.
